core/src/main/java/ai/timefold/solver/core/impl/phase/scope/AbstractStepScope.java

```diff
@@ -44,10 +46,23 @@ public void setBestScoreImproved(Boolean bestScoreImproved) {
         this.bestScoreImproved = bestScoreImproved;
     }
 
+    public void incrementMoveEvaluationCount(Move<?> move) {
+        if (isPhaseEnableCollectMetrics()) {
+            getPhaseScope().getSolverScope().addMoveEvaluationCount(1L);
+            if (getPhaseScope().getSolverScope().isMetricEnabled(SolverMetric.MOVE_COUNT_PER_TYPE)) {
+                getPhaseScope().getSolverScope().incrementMoveEvaluationCountPerType(move);
+            }
+        }
+    }
+
     // ************************************************************************
     // Calculated methods
     // ************************************************************************
 
+    public boolean isPhaseEnableCollectMetrics() {
+        return getPhaseScope().isEnableCollectMetrics();
+    }
```

Comment:
Ah, I see, it _is_ on the phase, you just make it available on the step like this.
I wouldn't - it increases amount of code in one place (here) to decrease amount of code later (at call time). But arguably, it adds more code than it removes.